hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/HtsSystemContract.java

```diff
@@ -70,6 +76,31 @@ private static FullResult resultOfExecuting(
         final HtsCall.PricedResult pricedResult;
         try {
             pricedResult = call.execute(frame);
+            final var proxyWorldUpdater = FrameUtils.proxyUpdaterFor(frame);
+            if (proxyWorldUpdater != null) {
```

Comment:
Yes, I would agree to that in a normal scenario but running `HtsSystemContractTest.java` without the check yields 2 failing tests because of `proxyWorldUpdater ` is null in these cases

<img width="410" alt="image" src="https://github.com/hashgraph/hedera-services/assets/39398927/299f90b8-27a6-471c-bb10-a25e7e1c5427">

So we should either adapt the tests or leave the check..
Your thoughts?

Reply:
Ah yes we should make those unit tests mock the `MessageFrame` as necessary to have a non-null updater here. 🙏